### PR TITLE
fix(api): Fix tracing TypeError for static and class methods

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
Fixes TypeError that occurred when static or class methods, which were passed in the `functions_to_trace` argument when initializing the SDK, were called on an instance. 

We also need to update docs to specify order when using `sentry_sdk.trace` decorator (as in the workaround I suggested on the issue)

Fixes GH-2525

### CODE REVIEW

File: tests/tracing/test_decorator_py3.py [hunk 1]
```diff
@@ -101,3 +118,33 @@ async def test_trace_decorator_async_py3_no_trx():
...
+        ".".join([TestClass.class_.__module__, TestClass.class_.__qualname__])
+        == test_classmethod_name
+    ), "The test class method was moved or renamed. Please update the name accordingly"
+
+    sentry_init(functions_to_trace=[{"qualified_name": test_classmethod_name}])
+
+    for instance_or_class in (TestClass, TestClass()):
+        with patch_start_child() as fake_start_child:
+            assert instance_or_class.class_(1) == (TestClass, 1)
+            fake_start_child.assert_called_once()
```
Comment (szokeasaurusrex) [inline tests/tracing/test_decorator_py3.py:150]: We should probably move these to `test_basics` since they are not specific to the test decorator and should also run on Python 2 (put them here originally when I thought I would fix the bug differently)

Comment (sentrivana) [reply tests/tracing/test_decorator_py3.py:150]: Fine with me either way

**Review by sentrivana — APPROVED** [review]
👍🏻

File: tests/tracing/test_decorator_async_py3.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+from unittest import mock
```
Comment (szokeasaurusrex) [inline tests/tracing/test_decorator_async_py3.py:1]: This file moved from `tests/tracing/test_decorator_py3.py`

**Comment by szokeasaurusrex** [comment]
I decided to move the tests I created to `test_basics.py`, and while doing so, I realized I could also refactor the tracing decorator tests to reuse common logic, rather than repeating extremely similar code snippets in multiple locations. I have added these improvements to this PR, and would appreciate a re-review.

File: sentry_sdk/integrations/aiohttp.py [hunk 3]
```diff
@@ -141,7 +141,7 @@ async def sentry_app_handle(self, request, *args, **kwargs):
                         transaction.set_http_status(response.status)
                         return response
 
-        Application._handle = sentry_app_handle  # type: ignore[method-assign]
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/aiohttp.py:144]: `mypy` was complaining about the `type: ignore` comments in this file, so I had to make changes here, even though the changes are unrelated to the rest of the PR

**Review by antonpirker — APPROVED** [review]
looks good
